x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_edit/common_rule_field_edit.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { DiffableCommonFields } from '../../../../../../../common/api/detection_engine';
+import { FieldFormWrapper } from './field_form_wrapper';
+import { NameEdit, nameSchema } from './fields/name';
+
+interface CommonRuleFieldEditProps {
+  fieldName: keyof DiffableCommonFields;
+}
+
+export function CommonRuleFieldEdit({ fieldName }: CommonRuleFieldEditProps) {
+  switch (fieldName) {
+    case 'name':
+      return <FieldFormWrapper component={NameEdit} fieldFormSchema={nameSchema} />;
+    case 'version':
```

Comment:
Do you know how to do this in a type-safe way? In case in the future a new field is added, we'd want TS to complain (`assertUnreachable`). To do this we'll need a TS type that has all the fields except the ones that are hidden.

Here's what I tried, but it didn't work for me.
```ts
const hiddenFields = new Set([
  'alert_suppression',
  'author',
  'rule_id',
  'license',
  'version',
] as const);

// Convert Set to an array to extract union type from it
const hiddenFieldsArray = [...hiddenFields];

// Extract the union type
type HiddenField = (typeof hiddenFieldsArray)[number];

type VisibleCommonField = Omit<keyof DiffableCommonFields, HiddenField>;

const a: VisibleCommonField = 'name'; // TS allows this
const b: VisibleCommonField = 'version'; // TS allows this, but shouldn't
```

Perhaps you might have a better ideas on how to implement this.